client/src/utils/userdata.ts

```diff
@@ -1,4 +1,44 @@
 // Functions to retrieve and manipulate data in local storage
 
-export const getUser: () => { hash?: string; username?: string } = () =>
-  JSON.parse(localStorage.getItem("user") || "{}");
+import type { Chats, Message, User } from "../type/global";
+
+enum Keys {
+  Chat = "chat",
+  User = "user",
+}
+
+export const getChats = () => {
+  return JSON.parse(localStorage.getItem(Keys.Chat) || "{}");
+};
+
+export const initChat = (chats: Chats, chatName: string): Chats => {
+  if (!chats[chatName]) {
+    chats[chatName] = [];
+    updateLocalKey(Keys.Chat, JSON.stringify(chats));
+  }
+
+  return chats;
+};
+
+const updateLocalKey = (item: Keys, value: string) => {
+  return localStorage.setItem(item, value);
```

Comment:
I suggest moving the JSON.stringify here
`localStorage.setItem(item, JSON.stringify(value));` instead of having the callers everywhere do so.

Reply:
I applied this change!